src/python/grpcio/grpc/_channel.py

```diff
@@ -263,35 +263,45 @@ def _rpc_state_string(class_name, rpc_state):
                 rpc_state.debug_error_string)
 
 
-class _RpcError(grpc.RpcError, grpc.Call):
+class _RpcError(grpc.RpcError, grpc.Call, grpc.Future):
     """An RPC error not tied to the execution of a particular RPC.
 
+    The state passed to _RpcError must be guaranteed not to be accessed by any
+    other threads.
```

Comment:
Can we use the copy-on-creation mode here? So the `RpcState` object being copied in the initialization function? In my mind, `_RpcError` is a data structure to store error info, it should be fine to keep a snapshot of the `RpcState`.